Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -1963,17 +1960,23 @@ extension SourceKitServer {
     }
     // For call hierarchy preparation we only locate the definition
     guard let usr = symbol.usr else { return nil }
-    return index.occurrences(ofUSR: usr, roles: [.definition, .declaration])
-      .compactMap { info -> CallHierarchyItem? in
-        guard let location = indexToLSPLocation(info.location) else {
-          return nil
-        }
-        return self.indexToLSPCallHierarchyItem(
-          symbol: info.symbol,
-          moduleName: info.location.moduleName,
-          location: location
-        )
-      }
+
+    // Only return a single call hierarchy item. Returning multiple doesn't make sense because they will all have the
+    // same USR (because we query them by USR) and will thus expand to the exact same call hierarchy.
+    // Also, VS Code doesn't seem to like multiple call hiearchy items being returned and fails to display any results
+    // if they are, failing with `Cannot read properties of undefined (reading 'map')`.
+    guard let definition = index.definitionOrDeclarationOccurances(ofUSR: usr).first else {
```

Comment:
Should we sort by location/something so this is stable?

Reply:
That applies to quite a few locations. I filed rdar://123487052 and will take care of it in a follow-up PR.